rust_src/src/fonts.rs

```diff
@@ -0,0 +1,64 @@
+use remacs_macros::lisp_fn;
+use remacs_sys::{PseudovecType, Qfont_spec, Qfont_entity, Qfont_object, Qsymbolp,
+                 wrong_type_argument};
+use lisp::{LispObject, Qnil};
+
+// See font_property_index in font.h for details.
+#[allow(non_camel_case_types, dead_code)]
+#[repr(C)]
+enum font_property_index {
```

Comment:
This enum will probably want to be defined in the remacs-sys crate instead of here. That crate is where we house the code related to FFI and representing structures that are defined in the C layer.